### PR TITLE
Das/v03

### PR DESCRIPTION
**Pull request by mezarque**
This PR adds the following features:
- built-in color vision deficiency testing for colors and images using [`colorspacious`](https://colorspacious.readthedocs.io/en/latest/tutorial.html#simulating-colorblindness)
- improved `arcadia:cividis` gradient
- strong monocolor gradients
- improved bicolor gradients

### CODE REVIEW

File: arcadia_pycolor/colors.py [hunk 1]
```diff
@@ -140,13 +145,18 @@
 The colors have also been optimized to be nearly-perceptually uniform based on lightness.
 """
 
-viridis = {
+viridis_v1 = {
```
Comment (mertcelebi) [inline arcadia_pycolor/colors.py:148]: what are the differences here?

File: arcadia_pycolor/functions.py [hunk 2]
```diff
@@ -369,3 +373,81 @@ def plot_color_lightness(cmap_dict: dict, title = None,
         return fig
     
     plt.show()
+    
+    
+def make_cvd_dict(form: str, severity: int):
```
Comment (mertcelebi) [inline arcadia_pycolor/functions.py:378]: awesome

File: arcadia_pycolor/functions.py [hunk 3]
```diff
@@ -6,12 +6,13 @@
 from plotly.subplots import make_subplots
 from IPython.display import Markdown, display
 import colorsys
-from colorspacious import cspace_converter
+from colorspacious import cspace_converter, cspace_convert
```
Comment (mertcelebi) [inline arcadia_pycolor/functions.py:9]: Are the functions in this file exposed to the end-user? Or there mostly for internal use? If the latter, `utils` maybe a better name

File: arcadia_pycolor/functions.py [hunk 4]
```diff
@@ -369,3 +373,81 @@ def plot_color_lightness(cmap_dict: dict, title = None,
...
+        rgb_colors = [mpl.colors.to_rgb(i) for i in colors]
+    else:
+        rgb_colors = colors
+    
+    cvd_rgb_colors = np.clip(cspace_convert(rgb_colors, cvd_space, "sRGB1"), 0, 1)
+    cvd_hex_colors = [mpl.colors.to_hex(color) for color in cvd_rgb_colors]
+
+    return cvd_hex_colors
+
+def simulate_cvd_image(file: str, severity = 100, fig_width = 10, show = True):
```
Comment (mertcelebi) [inline arcadia_pycolor/functions.py:410]: This seems more like for testing than not, but probably fine to have here

**Review by mertcelebi — CHANGES_REQUESTED** [review]
A couple questions + suggestions

File: arcadia_pycolor/colors.py (hunk 1 above)
Comment (mezarque) [reply arcadia_pycolor/colors.py:148]: I added the v1 version to keep as a deprecated version - the new "viridis" has a purple-ish color towards the end of the gradient. I can probably remove the v1 version now.

File: arcadia_pycolor/functions.py (hunk 3 above)
Comment (mezarque) [reply arcadia_pycolor/functions.py:9]: The plan was to expose these to the end-user; the functions listed under `__all__` are made available under the package namespace when using `import arcadia_pycolor` or `import arcadia_pycolor as apc`.

**Comment by mezarque** [comment]
Sorry for the long delay on getting to this! I went through and addressed the comments and also added some documentation + linting for the scripts.

**Review by mertcelebi — APPROVED** [review]
I am going to approve with a couple comments:
* Can you extend the README for setup instructions for people trying to improve on this locally? Like how to install the right packages etc?
* And make sure to proof-read the README?
* Is there a way to pin the underlying package versions?
* Can you add an MIT license?
* More discussions-y: for stuff like `simulate_cvd`, do you expect the average user to use those? or are there in the package for development reasons?

**Comment by mezarque** [comment]
- added a "For Developers" section and a "Contributing" section to the README
- added a YML for the conda env used to develop the package
- pinned package versions to YML and `setup.py`
- added MIT license
- `simulte_cvd` is meant for general users to check whether their figures are colorblind-friendly, yes. Something all scientists generating figures should probably be doing, but I think Publishing is doing that currently. Hopefully this helps streamline / centralize that

**Pull request merged**